### PR TITLE
perf:updated functions to pass R-CMD Check: no visible binding for gl…

### PR DESCRIPTION
**Pull request by valentina-buoro**
# Description
This is PR is a contribution for Outreachy internship.  It fixes the warning for 
`filter_function: no visible binding for global variable 'numeric_column_in_data'` in the following functions; lineage.neighbors.plot, lineage_sunburst, make_accnums_unique,make_df_iprscan_domains in #60 

## What kind of change(s) are included?

- [ ] Feature (adds or updates new capabilities)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] I have deleted all non-relevant text in this pull request template.
- [x] @jananiravi  please review these changes. Thank you

### CODE REVIEW

File: R/ipr2viz.R
Comment (the-mayer) [inline R/ipr2viz.R:1]: Looks like there are a few additional global variables to define for `ipr2viz()`

File: R/plotting.R [hunk 1]
```diff
@@ -1223,13 +1223,13 @@ lineage_sunburst <- function(prot, lineage_column = "Lineage",
         group_by_at(levels_vec) %>%
         summarise(size = n())
     protLevels <- protLevels %>% arrange()
-    tree <- d3_nest(protLevels, value_cols = "size")
+    tree <- .data$d3_nest(protLevels, value_cols = "size")
```
Comment (the-mayer) [inline R/plotting.R:1226]: This is actually looking for an additional import. Adding an Rd tag declaring this import to fix:
```
@importFrom d3r d3_nest
```

File: R/plotting.R [hunk 2]
```diff
@@ -1223,13 +1223,13 @@ lineage_sunburst <- function(prot, lineage_column = "Lineage",
...
 
     # Plot sunburst
     if (type == "sunburst") {
-        result <- sunburst(tree, legend = list(w = 225, h = 15, r = 5, s = 5), colors = cpcols, legendOrder = legendOrder, width = "100%", height = "100%")
+        result <- sunburst(tree, legend = list(w = 225, h = 15, r = 5, s = 5), colors = .data$cpcols, legendOrder = legendOrder, width = "100%", height = "100%")
     } else if (type == "sund2b") {
-        result <- sund2b(tree)
+        result <- .data$sund2b(tree)
```
Comment (the-mayer) [inline R/plotting.R:1232]: Similarly, this is looking for a function to be defined.

**Review by the-mayer — APPROVED** [review]
Thank you very much for your contributions to this issue, @valentina-buoro! 

I've made a few slight modifications -- please feel free to review at your leisure! Everything I changed was somewhat outside of the scope of the original issue. There are a few related warnings that end up being generated by the R-CMD check. 

I'm approving this PR and will merge shortly!

**Comment by valentina-buoro** [comment]
Thank you @the-mayer I'll review the changes that were made.

**Pull request merged**